Sources/MapboxNavigation/StatusView.swift

```diff
@@ -117,62 +151,111 @@ public class StatusView: UIControl {
         }
     }
     
+    /**
+     Shows the status view for a specified amount of time.
+     `showStatus()` uses a default value for priority and the title input as identifier. To use these variables, use `show(_:)`
+     */
+    @available(*, deprecated, message: "Add a status using show(_:) instead")
     public func showStatus(title: String, spinner spin: Bool = false, duration: TimeInterval, animated: Bool = true, interactive: Bool = false) {
-        show(title, showSpinner: spin, interactive: interactive)
-        guard duration < .infinity else { return }
-        hide(delay: duration, animated: animated)
+        let status = Status(identifier: title, title: title, spinner: spin, duration: duration, animated: animated, interactive: interactive, priority: StatusView.Priority(rawValue: 1))
+        show(status)
+    }
+    
+    /**
+     Adds a new status to statuses array.
+     */
+    func show(_ status: Status) {
+        if let row = statuses.firstIndex(where: {$0.identifier.contains(status.identifier)}) {
+            statuses[row] = status
+        } else {
+            statuses.append(status)
+        }
+        manageStatuses()
+    }
+    
+    /**
+     Manages showing and hiding Statuses and the status view itself.
+     */
+    func manageStatuses(status: Status? = nil) {
+        if statuses.isEmpty {
+            hide(delay: status?.duration ?? 0, animated: status?.animated ?? true)
+        } else {
+            // if we hide a Status and there are Statuses left in the statuses array, show the Status with highest priority
+            guard let highestPriorityStatus = statuses.min(by: {$0.priority.rawValue < $1.priority.rawValue}) else { return }
+            show(status: highestPriorityStatus)
+            hide(with: highestPriorityStatus, delay: highestPriorityStatus.duration)
+        }
+    }
+    
+    /**
+     Hides a given Status without hiding the status view.
+     */
+    func hide(_ status: Status?) {
+        guard let identifier = status?.identifier else { return }
+        guard let row = statuses.firstIndex(where: {$0.identifier.contains(identifier)}) else { return }
```

Comment:
This should be a simple `==` check now that identifiers are independent of labels.